src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -71,6 +71,13 @@ public synchronized void deleteTask(ReadOnlyTask target) throws TaskNotFoundExce
     }
 
     @Override
+    public synchronized ReadOnlyTask deleteTask(String target) throws TaskNotFoundException {
```

Comment:
If you can read, deleting the task with name argument. Method overloading. Alternative interfaces to do the same execution rather than executing outside the class of responsibility.